### PR TITLE
airlock: add "serve" command and TOML config

### PR DESCRIPTION
**Pull request by lucab**
This adds basic CLI and TOML config skeleton, landing an initial `airlock serve` command (whose logic will follow suit).

### CODE REVIEW

File: internal/cli/common.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	runSettings *config.Settings
+)
+
+// Init initializes the CLI environment for airlock
+func Init() (*cobra.Command, error) {
+	airlockCmd.PersistentFlags().StringVarP(&configPath, "config", "c", "/etc/airlock/config.toml", "path to configuration file")
+
+	airlockCmd.AddCommand(cmdServe)
+
+	logrus.SetLevel(logrus.DebugLevel)
```
Comment (arithx) [inline internal/cli/common.go]: Should we default to a higher log level and add a `verbose` flag now to enable debug logging?

File: internal/config/settings.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+	EtcdEndpoints  []string
+	EtcdTxnTimeout time.Duration
+
+	LockGroups map[string]uint64
+}
+
+// Parse parses a TOML configuration file and default values
+// into runtime settings.
+func Parse(fpath string) (*Settings, error) {
```
Comment (arithx) [inline internal/config/settings.go]: While returning a pointer makes the error return statement feel cleaner it doesn't seem like a `nil` return struct can ever be valid and we immediately assume that the returned struct isn't nil in `commonSetup` if an error isn't returned. Seems like changing it to return a non-pointer and just an empty object when an error occurs would work and reduce chances of hitting a nil-pointer dereference in the future.

File: internal/cli/common.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	}).Debug("etcd3 configuration")
+
+	return nil
+}
+
+func validateSettings(cfg config.Settings) error {
+	if len(cfg.EtcdEndpoints) == 0 {
+		return errors.New("no etcd3 endpoints configured")
+	}
+	if len(cfg.LockGroups) == 0 {
```
Comment (arithx) [inline internal/cli/common.go]: Can `LockGroups` ever actually have a length of 0? In `Parse` if it has a length of 0 the `default` key gets assigned to 1

File: internal/cli/common.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
+package cli
+
+import (
+	"errors"
+
+	"github.com/coreos/airlock/internal/config"
```
Comment (arithx) [inline internal/cli/common.go]: In Mantle & Ignition we're formatting our imports like so:

```
<stdlib>

<external lib>

<internal lib>
```

This doesn't seem to be a golang standard so not a big blocker but figured I'd bring it up.

File: internal/cli/common.go (hunk 1 above)
Comment (lucab) [reply internal/cli/common.go]: Added a counting `-v` flag. Its zero default now maps to `WarnLevel`.

File: internal/cli/common.go (hunk 3 above)
Comment (lucab) [reply internal/cli/common.go]: Not with the current logic, you are on spot. Though, I prefer to leave this sanity check in place to catch any possible future buggy rework around this.

**Comment by lucab** [comment]
@arithx thanks for reviewing! I have addressed all your comments and rebased.

File: internal/config/toml.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+// serviceSection holds the optional `service` fragment
+type serviceSection struct {
+	Address *string
+	Port    *uint64
+}
+
+// etcd3Section holds the optional `etcd3` fragment
+type etcd3Section struct {
+	Endpoints    []string
```
Comment (arithx) [inline internal/config/toml.go]: Do you see any value in adding toml struct tags to all values (even if the default behavior of lowercasing the field name is correct) for consistency when reading the code?

**Comment by lucab** [comment]
@arithx I've piled one commit on top, with field names, if you want to have a last look. I do agree it reads better. Do you have merge power in here?

**Pull request merged**